File: src/main/java/com/google/flourbot/macro/Macro.java
package com.google.flourbot.macro;

public class Macro {
    private String creatorId;
    private String macroName;
    private String docId;
    private Trigger macroTrigger;
    private Action macroAction;
    
    public Macro (String creatorId, String macroName, String docId, Trigger macroTrigger, Action macroAction) {
        this.creatorId = creatorId;
        this.macroName = macroName;
        this.docId = docId;
        this.macroTrigger = macroTrigger;
        this.macroAction = macroAction;
    }

    public String creatorId () {
        return creatorId;
    }

    public String getMacroName () {
        return macroName;
    }

    public String getDocId () {
        return docId;
    }
}

File: flutter_src/lib/model/entities/macro.dart
import 'package:equatable/equatable.dart';
import 'package:meta/meta.dart';

class Macro extends Equatable {

  String macroId;
  final String macroName;
  final String description;
  final String creatorId;
  final dynamic trigger;
  final dynamic action;

  Macro({
    @required this.macroName,
    @required this.description,
    @required this.creatorId,
    @required this.trigger,
    @required this.action,
  });

  @override
  List<Object> get props => [macroName, description, creatorId, trigger, action];
}
